optimum/exporters/onnx/__main__.py

```diff
@@ -381,86 +392,160 @@ def main_export(
+        if (
+            original_task == "auto" and not monolith
+        ):  # Make -with-past the default if --task was not explicitely specified
             task = task + "-with-past"
         else:
             logger.info(
                 f"The task `{task}` was manually specified, and past key values will not be reused in the decoding."
                 f" if needed, please pass `--task {task}-with-past` to export using the past key values."
             )
+            model.config.use_cache = False
 
-    if task.endswith("-with-past") and monolith is True:
-        task_non_past = task.replace("-with-past", "")
-        raise ValueError(
-            f"The task {task} is not compatible with the --monolith argument. Please either use"
-            f" `--task {task_non_past} --monolith`, or `--task {task}` without the monolith argument."
-        )
-
-    if original_task == "auto":
-        synonyms_for_task = sorted(TasksManager.synonyms_for_task(task))
-        if synonyms_for_task:
-            synonyms_for_task = ", ".join(synonyms_for_task)
-            possible_synonyms = f" (possible synonyms are: {synonyms_for_task})"
-        else:
-            possible_synonyms = ""
-        logger.info(f"Automatic task detection to {task}{possible_synonyms}.")
+    if task.endswith("with-past"):
+        model.config.use_cache = True
 
     # The preprocessors are loaded as they may be useful to export the model. Notably, some of the static input shapes may be stored in the
     # preprocessors config.
     preprocessors = maybe_load_preprocessors(
         model_name_or_path, subfolder=subfolder, trust_remote_code=trust_remote_code
     )
+
+    onnx_export(
+        model=model,
+        output=output,
+        opset=opset,
+        optimize=optimize,
+        monolith=monolith,
+        no_post_process=no_post_process,
+        atol=atol,
+        do_validation=do_validation,
+        model_kwargs=model_kwargs,
+        custom_onnx_configs=custom_onnx_configs,
+        fn_get_submodels=fn_get_submodels,
+        _variant=_variant,
+        legacy=legacy,
+        preprocessors=preprocessors,
+        device=device,
+        no_dynamic_axes=no_dynamic_axes,
+        task=task,
+        **kwargs_shapes,
+    )
+
+
+def onnx_export(
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    output: Union[str, Path],
+    opset: Optional[int] = None,
+    optimize: Optional[str] = None,
+    monolith: bool = False,
+    no_post_process: bool = False,
+    atol: Optional[float] = None,
+    do_validation: bool = True,
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    custom_onnx_configs: Optional[Dict[str, "OnnxConfig"]] = None,
+    fn_get_submodels: Optional[Callable] = None,
+    _variant: str = "default",
+    legacy: bool = False,
+    preprocessors: List = None,
+    device: str = "cpu",
+    no_dynamic_axes: bool = False,
+    task: Optional[str] = None,
```

Comment:
I was initially planning for `onnx_export` to not have any `task` argument and to infer it given the model but there is one main issue : some models like `XLMWithLMHeadModel` belongs to multiple [`MAPPING_NAMES`](https://github.com/huggingface/transformers/blob/v4.37.0/src/transformers/models/auto/modeling_auto.py#L467)